internal/app/frontend/frontend_service.go

```diff
@@ -354,7 +354,26 @@ func doWatchAssignments(ctx context.Context, id string, sender func(*pb.Assignme
 // AcknowledgeBackfill is used to notify OpenMatch about GameServer connection info.
 // This triggers an assignment process.
 func (s *frontendService) AcknowledgeBackfill(ctx context.Context, req *pb.AcknowledgeBackfillRequest) (*pb.Backfill, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+	if req.GetBackfillId() == "" {
+		return nil, status.Errorf(codes.InvalidArgument, ".BackfillId is required")
+	}
+	if req.GetAssignment().Connection == "" {
+		return nil, status.Errorf(codes.InvalidArgument, ".Assignment.Connection should be set")
+	}
+
+	bf, associatedTickets, err := s.store.GetBackfill(ctx, req.GetBackfillId())
+	if err != nil {
+		return nil, err
+	}
+	if len(associatedTickets) != 0 {
+		_, _, err = s.store.UpdateAssignments(ctx, &pb.AssignTicketsRequest{
+			Assignments: []*pb.AssignmentGroup{&pb.AssignmentGroup{TicketIds: associatedTickets, Assignment: req.GetAssignment()}},
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+	return bf, nil
```

Comment:
Seems AcknowledgeBackfill is missing?

Reply:
Added

Reply:
Curious if the entire logic (Assign Tickets, Remove the assigned ones from backfill, update acknowledged state on backfill) should live in the store layer itself? Having a AcknowledgeBackfill here that does all this individually and then calls AcknowledgeBackfill on the store which just updates the state is confusing. Any reason to have this logic separated from store? Is the store.AcknowledgeBackfill called other places where the rest of pieces of acknowledging are not needed?

Reply:
Actually I have chosen probably not the best name for statestore function `AcknowledgeBackfill`. I will rename statestore function to `UpdateAcknowledgmentTimetstamp`.

Reply:
Plan to do this in this PR or a followup PR?

Reply:
I have already applied this change to StateStore.